scripts/cicd/verify-cicd-operation.py

```diff
@@ -34,17 +38,54 @@ def runner(program, *args):
 logger.setLevel(logging.INFO)
 logger.addHandler(logging.handlers.SysLogHandler('/dev/log'))
 
-keyname = sys.argv[1] if len(sys.argv) >= 2 else "<none specified>"
+# The cluster_id argument is provided by the authorized_keys command entry for the ssh user's private key.
+# In other words, the ssh user cannot specify the cluster_id directly (they must have the correct private key
+# AND the key must be setup in the authorized_keys file appropriately).
+if len(sys.argv) != 2:
+    print("No cluster identifier specified")
+    sys.exit(12)
+
+cluster_id = sys.argv[1]
+
+# Sanity check the cluster_id from authorized_keys
+if not re.match("(^[a-zA-Z0-9][a-zA-Z0-9._-]+$)", cluster_id):
+    print("Invalid cluster identifier specified")
+    sys.exit(13)
+
 cmd = os.environ.get("SSH_ORIGINAL_COMMAND", "")
+args = cmd.split()
+
+if len(args) == 0:
+    print("No operation specified")
+    sys.exit(11)
+
+op = args.pop(0)  # Remove operation string from argument list and store
+
+match = re.match(r"(?P<operation>install|delete|upgrade|status|logs|perf1|perf2|perf3|build-ci-msg|smoketest)", op)
+
+if not match:
+    logger.info("%s Restricted key '%s' disallowed operation: %s" % (os.path.basename(__file__), cluster_id, op))
+    print("Operation doesn't match an allowed pattern ")
+    print("REJECTED ON HOST: " + socket.gethostname())
+    sys.exit(10)
+
+# Validate remaining arguments against strict patterns.
+# Only matched patters may be passed on to cicd-control. See approved list
+# at the top of this file.
+operation_args = []
+for s in args:
+    if re.match("(^docker-version=[a-zA-Z0-9.-_]+$)", s):  # Example docker version: docker-1.12.6-30.git97ba2c0.el7
```

Comment:
Sorry, missed this one.  That hyphen in the regex needs to be at the end, else you're taking all chars between . and _